airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/YamlSeedConfigPersistence.java

```diff
@@ -48,36 +48,45 @@
  */
 public class YamlSeedConfigPersistence implements ConfigPersistence {
 
+  public static Class<?> DEFAULT_SEED_DEFINITION_RESOURCE_CLASS = SeedType.class;
+
   private static final Map<AirbyteConfig, SeedType> CONFIG_SCHEMA_MAP = Map.of(
       ConfigSchema.STANDARD_SOURCE_DEFINITION, SeedType.STANDARD_SOURCE_DEFINITION,
       ConfigSchema.STANDARD_DESTINATION_DEFINITION, SeedType.STANDARD_DESTINATION_DEFINITION);
 
-  private static final YamlSeedConfigPersistence INSTANCE;
-  static {
-    try {
-      INSTANCE = new YamlSeedConfigPersistence();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-  }
+  private static final Object lock = new Object();
+  private static YamlSeedConfigPersistence INSTANCE;
 
   // A mapping from seed config type to config UUID to config.
   private final ImmutableMap<SeedType, Map<String, JsonNode>> allSeedConfigs;
 
-  private YamlSeedConfigPersistence() throws IOException {
+  private YamlSeedConfigPersistence(final Class<?> seedDefinitionsResourceClass) throws IOException {
     this.allSeedConfigs = ImmutableMap.<SeedType, Map<String, JsonNode>>builder()
-        .put(SeedType.STANDARD_SOURCE_DEFINITION, getConfigs(SeedType.STANDARD_SOURCE_DEFINITION))
-        .put(SeedType.STANDARD_DESTINATION_DEFINITION, getConfigs(SeedType.STANDARD_DESTINATION_DEFINITION))
+        .put(SeedType.STANDARD_SOURCE_DEFINITION, getConfigs(seedDefinitionsResourceClass, SeedType.STANDARD_SOURCE_DEFINITION))
+        .put(SeedType.STANDARD_DESTINATION_DEFINITION, getConfigs(seedDefinitionsResourceClass, SeedType.STANDARD_DESTINATION_DEFINITION))
         .build();
   }
 
+  public static void initialize(final Class<?> seedDefinitionsResourceClass) {
```

Comment:
I understand why you need to do that but this complex object lifecycle has some code smell.

Why is the YamlSeedConfigPersistence a singleton? Are we ever using it without injecting?

Reply:
Now that this class can be constructed differently, I think it should not be a singleton any more.

Reply:
cool. thanks. this was my intuition too.